mapconv.go

```diff
@@ -20,18 +21,22 @@ func ToMap(value interface{}, prefix string) (m map[string]string, err error) {
 		switch v.Kind() {
 		case reflect.Bool:
 			m[path] = strconv.FormatBool(v.Bool())
-		case reflect.Float64:
-			m[path] = strconv.FormatFloat(v.Float(), 'f', 0, 64)
-		case reflect.Int:
+		case reflect.Float32, reflect.Float64:
+			m[path] = strings.TrimRight(strings.TrimRight(strconv.FormatFloat(v.Float(), 'f', 6, 64), "0"), ".")
```

Comment:
I had some solutions in mind but, just put it off till later. So, thanks for the elegant solution.